python/tests/catalog/test_hive.py

```diff
@@ -0,0 +1,339 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+# pylint: disable=protected-access
+from unittest.mock import MagicMock
+
+import pytest
+from hive_metastore.ttypes import AlreadyExistsException
+from hive_metastore.ttypes import Database as HiveDatabase
+from hive_metastore.ttypes import (
+    FieldSchema,
+    InvalidOperationException,
+    MetaException,
+    NoSuchObjectException,
+    SerDeInfo,
+    SkewedInfo,
+    StorageDescriptor,
+)
+from hive_metastore.ttypes import Table as HiveTable
+
+from pyiceberg.catalog.hive import HiveCatalog
+from pyiceberg.exceptions import (
+    AlreadyExistsError,
+    NamespaceNotEmptyError,
+    NoSuchNamespaceError,
+    NoSuchTableError,
+)
+from pyiceberg.schema import Schema
+
+HIVE_CATALOG_NAME = "hive"
+HIVE_METASTORE_FAKE_URL = "thrift://unknown:9083"
+
+
+def test_create_table(table_schema_simple: Schema):
+    catalog = HiveCatalog(HIVE_CATALOG_NAME, {}, url=HIVE_METASTORE_FAKE_URL)
+
+    catalog._client = MagicMock()
+    catalog._client.create_table.return_value = None
+
+    catalog._client = MagicMock()
+    catalog._client.load_table.return_value = HiveTable(
```

Comment:
What is this testing? `load_table` isn't called in this test, and there's no validation for `create_table` so it doesn't appear to test anything. Can you get the object that was passed to `create_table` and verify it?

Reply:
In hive when we call the `create_table` we don't get the HiveTable back, therefore we get the table right after creating it. I've added the `assert_called_with` to validate the call. This will be further extended after implementing the `FileIO` for writing the JSON.